In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
from scipy import stats

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

Buyer Ratio.mtw


In [9]:
buyer_data = pd.read_csv('BuyerRatio.csv')
buyer_data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

### Hypothesis Formulation

* HO : All proportions are equal
* Ha : Not all proportions are equal.

#### Level of significance = 5%

**the dependent and independent variables are discrete**
* **So we will go for CHi- Square Test**

### Chi- Square Test

In [24]:
chi2_score,p_val,dof,expected = stats.chi2_contingency([buyer_data["East"], buyer_data["West"], buyer_data["North"], buyer_data["South"]])
print("**********************************************")
print('CHI SQUARE TEST RESULT')

print('chi2 value     :' ,chi2_score)
print('p_value        :' ,p_val)
print('dof            :' ,dof)
print('\n')
print('expected table :' ,expected)

**********************************************
CHI SQUARE TEST RESULT
chi2 value     : 1.5959455386610577
p_value        : 0.6603094907091882
dof            : 3


expected table : [[  42.76531299  442.23468701]
 [ 146.81287862 1518.18712138]
 [ 131.11756787 1355.88243213]
 [  72.30424052  747.69575948]]


In [27]:
if p_val<0.05:
    print('We reject the null hypothesis (H0) and we can say that Not All Proprtions are EQUAL')
else:
    print('We do not reject the null hypothesis (H0) and we can say that ALL Proportions are EQUAL.')

We do not reject the null hypothesis (H0) and we can say that ALL Proportions are EQUAL.


* **p =0.6603 >0.05  , it means male-female buyer rations are similar across all regions**.